src/pages/CommandView/commandViewHelpers.tsx

```diff
@@ -0,0 +1,244 @@
+      <Button onClick={() => navigate(link, { replace: true })}>
+        GO TO {`${message}`}
+      </Button>
+    </Box>
+  )
+}
+/*
+    Sends us to a useful page if we came to CommandView using the wrong method.
+ */
+const checkContext = (
+  namespace: string | undefined,
+  systemName: string | undefined,
+  version: string | undefined,
+  commandName: string | undefined,
+  system: StrippedSystem | undefined,
+  command: AugmentedCommand | undefined,
+): JSX.Element | undefined => {
+  if (!namespace) {
+    return <Navigator message={'systems'} link={'/systems'} />
+  }
+  if (!systemName) {
+    return (
+      <Navigator
+        message={`systems : ${namespace}`}
+        link={`/systems/${namespace}`}
+      />
+    )
+  }
+  if (!version || !commandName || !system || !command) {
+    return (
+      <Navigator
+        message={`systems : ${namespace} : ${systemName}`}
+        link={`/systems/${namespace}/${systemName}`}
+      />
+    )
+  }
+
+  return undefined
+}
+
+/**
+ * Determine if a Parameter is a Bytes parameter or, in the case of
+ * Dictionaries, if any sub-parameter is a Bytes parameter
+ *
+ * @param parameter - the Parameter to check
+ * @returns true or false
+ */
+const isByte = (parameter: Parameter): boolean => {
+  return (
+    parameter.type === 'Bytes' ||
+    (parameter.parameters.length > 0
+      ? isByteCommand(parameter.parameters)
+      : false)
+  )
+}
+
+/**
+ * Determine if any of the parameters of a command are bytes parameters
+ *
+ * @param parameters - the parameter array to check
+ * @returns true or false
+ */
+const isByteCommand = (parameters: Parameter[]) => {
+  return parameters.map(isByte).some((x) => x)
+}
+
+/**
+ * Handle resetting the model to initial values by retaining only
+ * bytes parameters values.
+ *
+ * @param initialData - the empty version of the model
+ * @param model - the updated version of the model
+ * @param parameters - the parameters of the command
+ * @returns an empty model that retains the values for bytes parameters
+ */
+const handleByteParametersReset = (
+  initialData: ObjectWithStringKeys,
+  model: ObjectWithStringKeys,
+  parameters: Parameter[],
+) => {
+  if ('parameters' in model) {
+    const updatedInitialData = { ...initialData }
+    const { parameters: updatedParameters } = updatedInitialData as {
+      parameters: ObjectWithStringKeys
+    }
+    const modelParameters = model['parameters'] as
+      | ObjectWithStringKeys
+      | undefined
+
+    if (modelParameters) {
+      for (const parameter of parameters) {
+        if (!isByte(parameter)) continue
+
+        const { key } = parameter
+
+        if (key in modelParameters) {
+          updatedParameters[key] = modelParameters[key]
+        }
+      }
+
+      return {
+        ...initialData,
+        parameters: updatedParameters,
+      }
+    }
+  }
+
+  return initialData
+}
+
+/* Pull the filename out of a dataURL string */
+const cleanUrlData = (urlData: string) => {
```

Comment:
This is so the display of the model in the UI only shows the filename of the chosen file (for Bytes parameters) rather than the whole dataUrl string.